server/api.ts

```diff
@@ -180,6 +182,41 @@ router.postAsync("/register", ensure.loggedIn, async (req, res) => {
     );
     return res.status(400).send({ error: `Your country can't participate in this division` });
   }
+  
+  if (tourney.discordServerId) {
+    if (!req.user.discordId) {
+      logger.info(`${req.user.username} failed to register for ${req.body.tourney} (discord account not linked)`);
+      return res.status(400).send({ error: "You have not linked your Discord account." });
```

Comment:
can we include some text to tell the user how they can link their discord account? (or ideally open user settings automatically)

Reply:
mmm I don't really like to open something automatically without context, so maybe an idea could be to show a popup message instead? something like, "Please link your Discord account in your user settings and then try again"

Reply:
it doesn't have to open the settings modal, just a little more detail in that error message to tell the user where they should go to link their discord would probably be helpful